web/src/pages/indicators/[slug]/history/index.page.tsx

```diff
@@ -0,0 +1,203 @@
+						{hasOnThisPageMenu ? (
+							<GridItem cols={12} md={4} lg={3}>
+								<OnThisPage sections={groupSections} />
+							</GridItem>
+						) : null}
+						<GridItem
+							cols={12}
+							md={hasOnThisPageMenu ? 8 : 12}
+							lg={hasOnThisPageMenu ? 9 : 12}
+						>
+							<ResourceList groups={project.groups} />
+						</GridItem>
+					</Grid>
+				</GridItem>
+			</Grid>
+		</>
+	);
+}
+
+export const getServerSideProps: GetServerSideProps<
+	HistoryPageProps,
+	{ slug: string }
+> = async ({ params, resolvedUrl }) => {
+	const result = await validateRouteParams(params, resolvedUrl);
+
+	if ("notFound" in result || "redirect" in result) return result;
+
+	const {
+		project,
+		historyPanels,
+		product,
+		productPath,
+		hasEvidenceResources,
+		hasInfoForPublicResources,
+		hasToolsAndResources,
+		hasHistory,
+	} = result;
+
+	if (!project) return { notFound: true };
+
+	if (!hasHistory) return { notFound: true };
+
+	const groups = historyPanels.sort(byTitleAlphabetically).map((panel) => {
+		const indevResource =
+			panel.embedded.niceIndevResourceList.embedded.niceIndevResource;
+
+		const indevResources = Array.isArray(indevResource)
+			? indevResource
+			: [indevResource];
+
+		const subGroups: ResourceSubGroupViewModel[] = [];
+
+		let currentSubGroup: ResourceSubGroupViewModel;
+
+		indevResources.forEach((resource) => {
+			if (resource.textOnly) {
+				currentSubGroup = { title: resource.title, resourceLinks: [] };
+				subGroups.push(currentSubGroup);
+			} else {
+				if (!currentSubGroup) {
+					currentSubGroup = { title: panel.title, resourceLinks: [] };
+					subGroups.push(currentSubGroup);
+				}
+
+				const { mimeType, length, resourceTitleId, fileName } =
+						resource.embedded.niceIndevFile,
+					isHTML = mimeType === "text/html",
+					fileSize = isHTML ? null : length,
+					fileTypeName = isHTML ? null : getFileTypeNameFromMime(mimeType),
+					href = isHTML
+						? `${productPath}/history/${resourceTitleId}`
+						: `${productPath}/history/downloads/${
+								product.id
+						  }-${resourceTitleId}.${fileName.split(".").slice(-1)[0]}`;
+
+				currentSubGroup.resourceLinks.push({
+					title: resource.title,
+					href,
+					fileTypeName,
+					fileSize,
+					date: resource.publishedDate,
+					type: panel.title,
+				});
+			}
+		});
+
+		return {
+			title: panel.title,
+			subGroups,
+		};
+	});
+
+	return {
+		props: {
+			productPath,
+			hasEvidenceResources,
+			hasInfoForPublicResources,
+			hasToolsAndResources,
+			hasHistory,
+			product: {
+				id: product.id,
+				title: product.title,
+				productTypeName: product.productTypeName,
+				publishedDate: product.publishedDate,
+				lastMajorModificationDate: product.lastMajorModificationDate,
+			},
+			project: {
+				reference: project.reference,
+				title: project.title,
+				groups,
+			},
+			groupSections: groups.map(({ title }, i) => ({
+				slug: title.replace(/ /g, "-").toLowerCase() + i,
```

Comment:
Why don't we use slugify from utils here?

Reply:
Good shout